server/punishments.js

```diff
@@ -1653,48 +1576,99 @@ const Punishments = new (class {
 
 		return punishments;
 	}
+	/**
+	 * @param {string} [roomid]
+	 * @param {boolean} [ignoreMutes]
+	 */
+	getPunishments(roomid, ignoreMutes) {
+		/** @type {Map<string, PunishmentEntry>} */
+		const punishmentTable = new Map();
+		if (roomid && (!Punishments.roomIps.has(roomid) || !Punishments.roomUserids.has(roomid))) return punishmentTable;
+		// @ts-ignore
+		(roomid ? Punishments.roomIps.get(roomid) : Punishments.ips).forEach((punishment, ip) => {
```

Comment:
why is this a `forEach` instead of just a normal `for` loop? Don't our contributing guidelines discourage `forEach` (yes, I know it was a `forEach` before)?

Reply:
This isn't actually a `forEach` loop - it's just `PunishmentMap#forEach` which uses a `for...of` loop under the hood.